tests/parametric/test_config_consistency.py

```diff
@@ -0,0 +1,52 @@
+"""
+Test configuration consistency for functions among different languages for APM.
+"""
+import pytest
+
+from utils import scenarios
+
+parametrize = pytest.mark.parametrize
+
+TEST_SERVICE = "test_service"
+TEST_ENV = "test_env"
+DEFAULT_ENVVARS = {
+    "DD_SERVICE": TEST_SERVICE,
+    "DD_ENV": TEST_ENV,
+}
+
+@scenarios.parametric
+class TestTraceEnabled:
+    @parametrize(
+        "library_env", [{**DEFAULT_ENVVARS, "DD_TRACE_ENABLED": "true"},],
+    )
+    def test_tracing_enabled(self, library_env, test_agent, test_library):
+        trace_enabled_env = library_env.get("DD_TRACE_ENABLED") == "true"
+        if trace_enabled_env:
+            with test_library:
+                with test_library.start_span("allowed"):
+                    pass
+            test_agent.wait_for_num_traces(num=1, clear=True)
+            assert (
+                True
+            ), "DD_TRACE_ENABLED=true and wait_for_num_traces does not raise an exception after waiting for 1 trace."
+        else:
+            assert False, f"Assertion failed: expected true, but got " + str(trace_enabled_env)
+
+    @parametrize(
+        "library_env", [{**DEFAULT_ENVVARS, "DD_TRACE_ENABLED": "false"},],
+    )
+    def test_tracing_disabled(self, library_env, test_agent, test_library):
+        trace_enabled_env = library_env.get("DD_TRACE_ENABLED") == "false"
+        if trace_enabled_env:
+            with test_library:
+                with test_library.start_span("allowed"):
+                    pass
+            with pytest.raises(ValueError):
+                test_agent.wait_for_num_traces(num=1, clear=True)
+
```

Comment:
Hmm.. the test is kind of confusing this way. I'd maybe change the format of this test to something like:
- assert DD_TRACE_ENABLED is false
- Start a span, then assert 0 traces (or, like we discussed, assert wait_for_num_traces(num=1) reports an error)

Reply:
Is it excessive to assert that `DD_TRACE_ENABLED` is false especially because that condition is checked for in order to run these lines of code? Would it be better for test visibility purposes to scrap the if statement verifying the value of `DD_TRACE_ENABLED` and replace it with an assertion?


Something like this:

```suggestion
def test_tracing_disabled(self, library_env, test_agent, test_library):
  assert library_env.get("DD_TRACE_ENABLED") == "false"
  with test_library:
      with test_library.start_span("allowed"):
          pass
  with pytest.raises(ValueError):
      test_agent.wait_for_num_traces(num=1, clear=True)
  assert (
      True
  ), "wait_for_num_traces raises an exception after waiting for 1 trace."  # wait_for_num_traces will throw an error if not received within 2 sec, so we expect to see an exception
  \
```